script/setup/install-cni

```diff
@@ -20,8 +20,17 @@
 # The commit defined in go.mod
 #
 set -eu -o pipefail
+GOMODEnv=`go env GOMOD`
+NULLValue=/dev/null
+if [ "$NULLValue" == "$GOMODEnv" ];then
+    echo "go.mod file not find"
+    exit 1
+fi
+
+CNI_COMMIT=${1:-$(grep containernetworking/plugins $GOMODEnv | awk '{print $2}')}
+
+echo "cni plugins version:"$CNI_COMMIT
```

Comment:
I think `go list` should be able to provide this information (without much of the "hackery pokery";

```bash
go list -f "{{.Version}}" -m github.com/containernetworking/plugins
v1.1.1
```

so something like;

```bash
CNI_COMMIT=${1:-$(go list -f "{{.Version}}" -m github.com/containernetworking/plugins)}
```

Reply:
OK, so with the Dockerfile stage looking like this (making sure we're not inside `/go`, otherwise it will error with `$GOPATH/go.mod exists but should not`:

```dockerfile
FROM golang AS cni
ARG DESTDIR=/build
WORKDIR /
COPY script/setup/install-cni go.mod ./
RUN ./install-cni
```

And the `script/setup/install-cni` script looking like this;

```bash
# ...
set -eu -o pipefail

CNI_COMMIT=${1:-$(go list -f "{{.Version}}" -m github.com/containernetworking/plugins)}
CNI_DIR=${DESTDIR:=''}/opt/cni
CNI_CONFIG_DIR=${DESTDIR}/etc/cni/net.d
```

At least the Dockerfile works;

```bash
docker build -t foo --target=cni -f ./contrib/Dockerfile.test .
# [+] Building 1.5s (8/8) FINISHED
# ...
# => => naming to docker.io/library/foo:latest                               0.0s
# => => unpacking to docker.io/library/foo:latest                            0.8s

docker run --rm foo ls -lR /build
/build:
total 8
drwxr-xr-x 3 root root 4096 Oct  8 14:20 etc
drwxr-xr-x 3 root root 4096 Oct  8 14:20 opt

/build/etc:
total 4
drwxr-xr-x 3 root root 4096 Oct  8 14:20 cni

/build/etc/cni:
total 4
drwxr-xr-x 2 root root 4096 Oct  8 14:20 net.d

/build/etc/cni/net.d:
total 4
-rw-r--r-- 1 root root 604 Oct  8 14:20 10-containerd-net.conflist

/build/opt:
total 4
drwxr-xr-x 3 root root 4096 Oct  8 14:20 cni

/build/opt/cni:
total 4
drwxr-xr-x 2 root root 4096 Oct  8 14:20 bin

/build/opt/cni/bin:
total 65312
-rwxr-xr-x 1 root root  3860602 Oct  8 14:20 bandwidth
-rwxr-xr-x 1 root root  4254984 Oct  8 14:20 bridge
-rwxr-xr-x 1 root root 10065455 Oct  8 14:20 dhcp
-rwxr-xr-x 1 root root  4442257 Oct  8 14:20 firewall
-rwxr-xr-x 1 root root  3881772 Oct  8 14:20 host-device
-rwxr-xr-x 1 root root  3322252 Oct  8 14:20 host-local
-rwxr-xr-x 1 root root  4045602 Oct  8 14:20 ipvlan
-rwxr-xr-x 1 root root  3396678 Oct  8 14:20 loopback
-rwxr-xr-x 1 root root  4054086 Oct  8 14:20 macvlan
-rwxr-xr-x 1 root root  3776569 Oct  8 14:20 portmap
-rwxr-xr-x 1 root root  4087947 Oct  8 14:20 ptp
-rwxr-xr-x 1 root root  3582996 Oct  8 14:20 sbr
-rwxr-xr-x 1 root root  2881418 Oct  8 14:20 static
-rwxr-xr-x 1 root root  3486219 Oct  8 14:20 tuning
-rwxr-xr-x 1 root root  4043797 Oct  8 14:20 vlan
-rwxr-xr-x 1 root root  3660834 Oct  8 14:20 vrf
```

Reply:
nice！I was thinking that we can use go.mod path as argument to parse.

Reply:
I don't think the usage of `install-cni` should be changed at the moment, it's not recommended to use `go.mod` path as a parameter; as suggested by @thaJeztah I think it's fine.